packages/atlas/src/components/_ypp/DetailsFormDialog/DetailsFormDialog.tsx

```diff
@@ -0,0 +1,125 @@
+import { useApolloClient } from '@apollo/client'
+import debouncePromise from 'awesome-debounce-promise'
+import { FC, useRef, useState } from 'react'
+import { Controller, useForm } from 'react-hook-form'
+
+import { GetBasicChannelsQuery, GetBasicChannelsQueryVariables } from '@/api/queries/__generated__/channels.generated'
+import { BasicChannelFieldsFragment } from '@/api/queries/__generated__/fragments.generated'
+import { GetMembershipsDocument } from '@/api/queries/__generated__/memberships.generated'
+import { Avatar, AvatarProps } from '@/components/Avatar'
+import { FormField } from '@/components/_inputs/FormField'
+import { Input } from '@/components/_inputs/Input'
+import { Loader } from '@/components/_loaders/Loader'
+import { EMAIL_PATTERN } from '@/config/regex'
+import { useAsset } from '@/providers/assets/assets.hooks'
+import { SentryLogger } from '@/utils/logs'
+
+import { FormFieldsWrapper } from './DetailsFormDialog.styles'
+
+export const DetailsFormDialog: FC = () => {
+  const [foundChannel, setFoundChannel] = useState<BasicChannelFieldsFragment | null>()
+  const [loading, setLoading] = useState(false)
+  const client = useApolloClient()
+  const {
+    setValue,
+    register,
+    control,
+    formState: { errors },
+  } = useForm<{ email: string | undefined; channel: string | undefined }>({ mode: 'onChange' })
+
+  const debounceFetchChannels = useRef(
+    debouncePromise(async (value?: string) => {
+      if (!value) {
+        return
+      }
+      try {
+        const {
+          data: { channels },
+        } = await client.query<GetBasicChannelsQuery, GetBasicChannelsQueryVariables>({
+          query: GetMembershipsDocument,
+          variables: { where: { title_eq: value } },
+        })
+        setFoundChannel(channels.length ? channels[0] : undefined)
+      } catch (error) {
+        SentryLogger.error('Failed to fetch memberships', 'WhiteListTextField', error)
+      } finally {
+        setLoading(false)
+      }
+    }, 500)
+  )
+  return (
+    <FormFieldsWrapper>
+      <FormField
+        label="Contact email"
+        description="We need your email address to send you payment information. No spam or marketing materials."
+        error={errors.email?.message}
+      >
+        <Controller
```

Comment:
Why do we need a controller here? We could use `register` as well if I'm not mistaken.